davidpil2002/sonic-mgmt
# Added Password Hardening tests cases

### Description of PR


Summary:
Added Password Hardening Tests according to HLD Test Plan

Context: By enabling the new Password Hardening feature and configured policies, the user will be enforce to use "strong" passwords according to the policies configured.

The new tests cases will cover all the possible password policies with good and bad cases, including:
* Perform show command
* Perform config command
* Verify default values of the feature
* Enable/Disable the feature
* Test all passw policies available:
  * Configure passw policy classes of every existing types
  * Configure passw policy  min using different valid values
  * Configure passw policy  age expiration using different valid values
  * Configure passw policy  age warning using different valid values
  * Configure passw policy  username-password-match
  * Configure passw policy  history using different valid values
Note
After creating a new policy is necessary to set a new password for a user to verify that the policy matches the configured.
The set configuration should be validated using the show command.
### Type of change



- [*] Test case(new/improvement)


### Back port request

### Approach
#### What is the motivation for this PR?
Add new tests cases for testing the feature Password Hardening.

#### How did you do it?
There are 10 test scenarios each one of them config different policies and test the case with good(strong) passwords and bad(weak) passwords expecting results according to the configuration and the flow.
The configuration is done by using sonic-cli, sending different "config" commands.
In addition, the test is testing that the configuration in PAM files in the system change according to the expected.
The expected files were saved in the directory "sample".

#### How did you verify/test it?
New tests cases passed
#### Any platform specific information?
N/A
#### Supported testbed topology if it's a new test case?
any
### Documentation

Link of HLD including feature design and Test plan:
https://github.com/Azure/SONiC/blob/master/doc/passw_hardening/hld_password_hardening.md
